Managers/PhaseManager.gd

```diff
@@ -12,6 +12,8 @@ signal on_phase_changed(new_phase: GlobalEnums.Phase, old_phase: GlobalEnums.Pha
 signal on_event_win
 ## When the player is dead (reduced to 0 health)
 signal on_defeat
+## When the phase is changed to PLAYER_ATTACKING
+signal on_turn_start
```

Comment:
there is already `on_phase_changed` which gives you the current phase of the combat. This is basically recreating a sub-case of a signal that already exists, and it might be confusing in the long run